### PR TITLE
HCA ESR/EE Connectivity

### PR DESCRIPTION
**Pull request by markolson**
Refactor of the existing Online Application code so that we can re-use some common connectivity code to connect with the Enrollment Service. This shouldn't be merged until we update the dev/staging/prod settings.yml files to account for the new keys.

### CODE REVIEW

**Comment by markolson** [comment]
Looks like I need to get settings.yml updated for ENV=test before this'll pass

File: lib/hca/configuration.rb [hunk 1]
```diff
@@ -57,6 +56,7 @@ def connection
         conn.options.timeout = 15       # TODO(molson): Make a config/setting
         conn.request :soap_headers
         conn.response :soap_parser
+        # conn.request(:curl, ::Logger.new(STDOUT), :warn)
```
Comment (lihanli) [inline lib/hca/configuration.rb]: is this still needed?

**Comment by ayaleloehr** [comment]
@markolson what's the status here?

**Comment by markolson** [comment]
Clearly a lot of merge conflicts to resolve, but this code was held up for a while waiting for some environment variables to get set, and then fell by the wayside. It'll be needed (maybe?) for HCA Prefill for the 1010EZR, but could do with a lot of cleanup before that point. Might be best to close and start over, and just re-use the existing environment variables.

**Comment by jkassemi** [comment]
@lihanli, @rlbaker - can you take a quick look at this and determine how we should proceed or close this out? Will close next week if there's no further action.

**Comment by lihanli** [comment]
i don't think this is needed anymore

**Pull request closed without merging**